### PR TITLE
Improve performance metrics API

### PR DESCRIPTION
**Pull request by imakhlin**
* Usability
* Performance
* Control

### CODE REVIEW

File: test/benchmark/BenchmarkIngest_test.go [hunk 1]
```diff
@@ -109,24 +109,28 @@ func BenchmarkIngest(b *testing.B) {
...
 	b.StartTimer()
-	for i := 0; i < b.N; i++ {
-		rowsAdded, err := runTest(i, appender, timestamps, sampleTemplates, refs,
-			testConfig.AppendOneByOne, testConfig.BatchSize, testConfig.Verbose, testConfig.ValidateRawData)
 
-		if err != nil {
-			b.Fatal(err)
-		}
+	var ec int
+	withTimer(b, fmt.Sprintf("BenchmarkIngest-%d -> Table: %s", b.N, tsdbPath), func() {
```
Comment (talIguaz) [inline test/benchmark/BenchmarkIngest_test.go]: Why do you need both `b.StartTimer() / b.StopTimer()` and `withTimer`?

File: internal/pkg/performance/metrics.go [hunk 2]
```diff
@@ -103,36 +109,33 @@ func (mr *MetricReporter) Stop() error {
...
+		body()
 	}
 }
 
-func (mr *MetricReporter) GetCounter(name string) (metrics.Counter, error) {
-	if mr.running {
-		return metrics.GetOrRegisterCounter(name, mr.registry), nil
-	} else {
-		return nil, errors.Errorf("failed to create counter '%s'. Reason: metric reporter in not running", name)
+func (mr *MetricReporter) IncrementCounter(name string, count int64) {
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply internal/pkg/performance/metrics.go]: It does not make it complex at all but provide you ability to add/subtract any number to/from the existing counter. In some cases you may want to add number of "pending samples" for example

File: pkg/appender/appender.go [hunk 3]
```diff
@@ -139,6 +140,8 @@ func NewMetricsCache(container *v3io.Container, logger logger.Logger, cfg *confi
 	newCache.newUpdates = make(chan int, 1000)
 
 	newCache.NameLabelMap = map[string]bool{}
+	newCache.performanceReporter = performance.ReporterInstanceFromConfig(cfg)
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply pkg/appender/appender.go]: reporter is also singleton, once created it always return ref. to the same instance

File: test/benchmark/BenchmarkIngest_test.go (hunk 1 above)
Comment (imakhlin) [reply test/benchmark/BenchmarkIngest_test.go]: It measures a test cycle time without using performance metrics (simple calc)
Example output:
```
BenchmarkIngest_test.go:315: BenchmarkIngest-200 -> Table: tsdb-BenchmarkIngest-200-2018-10-04T15_41_18_03_00
			Elapsed time: 3438 millisecond
```

**Pull request merged**